TACC/Frontera-Portal
# task/FP-427 - metadata replacement port

## Overview: ##

Replace metadata with a Django model for categories and app entires

## PR Status: ##


## Related Jira tickets: ##

* [FP-427](https://jira.tacc.utexas.edu/browse/FP-427)

## Summary of Changes: ##

- Add a new backend endpoint /api/workspace/tray
- Add a management command `python manage.py import-apps`
- Add admin interfaces for App Category and App Entry at /core/admin
- Refactor reducers/sagas to construct state.apps.categoryDict from the response from /api/workspace/tray

## Testing Steps: ##
1. In the docker container, run `python manage.py migrate`, then `python manage.py import-apps` - this should import all existing Frontera metadata into the App Tray Model
2. You may edit app search parameters in /core/admin

## Notes: ##
- There appear to be some invalid metadata causing some inconsistencies during the import-apps process. This results in matlab specifying a 'matlab' icon and vis-portal specifying a 'vis-portal' icon. These are non-existent and will result in a blank icon when viewed in Applications. This can be corrected via the management interface simply by deleting the contents of the 'icon' field in those App Tray entries
- I didn't delete the existing Apps backend views
- This port does less than the CEPv1 version. Namely, it does not attempt to pre-cache app definitions because that results in a very slow portal load. 
- Future work can involve refactoring Applications to support app stacking, app definition caching, etc.
